frontstage_api/common/request_handler.py

```diff
@@ -1,43 +1,28 @@
 import logging
 
 import requests
-from requests.exceptions import ConnectionError, ConnectTimeout
+from requests.exceptions import ConnectionError
 from structlog import wrap_logger
 
-from frontstage_api.exceptions.exceptions import FailedRequest, InvalidRequestMethod
+from frontstage_api.exceptions.exceptions import ApiError, InvalidRequestMethod
 
 
 logger = wrap_logger(logging.getLogger(__name__))
 
 
-def request_handler(method, url, headers=None, json=None, fail=True):
+def request_handler(method, url, auth=None, headers=None, json=None, error_code=None):
     try:
         if method == 'GET':
-            response = requests.get(url, headers=headers)
+            response = requests.get(url, auth=auth, headers=headers)
         elif method == 'POST':
-            response = requests.post(url, headers=headers, json=json)
+            response = requests.post(url, auth=auth, headers=headers, json=json)
         elif method == 'PUT':
-            response = requests.put(url, headers=headers, json=json)
+            response = requests.put(url, auth=auth, headers=headers, json=json)
         else:
             logger.error('Invalid request method', method=str(method), url=url)
             raise InvalidRequestMethod(method, url)
-    except ConnectTimeout as e:
-        logger.error('Connection to remote server timed out', method=method, url=url, exception=str(e))
-        if fail:
-            raise FailedRequest(method, url, e)
-        else:
-            return {
-                "error": "Connection timeout",
-                "url": url
-            }
     except ConnectionError as e:
```

Comment:
Do we need to catch the ConnectionError and wrap it in an ApiError or can we just catch with @api.errorhandler(ConnectionError)

Reply:
I was trying to return all errors in the same way from the app (generating them with ApiError). If we're just catching ConnectionError then we lose any reference to the error code associated with it, (assuming that we're happy with the error code method).